dev-utils/visualise-ui-hierarchy/Makefile

```diff
@@ -0,0 +1,46 @@
+CWD := $(shell pwd)
+POLYPOD_FEATURES_PATH := $(CWD)/../../features
+RCH := node $(CWD)/react-component-hierarchy/bin/rch.js -e
+
+.PHONY: all
+all: polyExplorer.svg facebookImport.svg
+
+.PHONY: rebuild
+rebuild: clean-files all
+
+%.svg: %.dot
+	fdp -T svg -o $@ $<
+
+.INTERMEDIATE: %.dot
+%.dot: %.txt
+	./tree2dot < $< > $@
+
+.INTERMEDIATE: polyExplorer.txt
+polyExplorer.txt: ensure-rch
+	cd $(POLYPOD_FEATURES_PATH)/polyExplorer; \
+		$(RCH) src/polyExplorer.jsx > $@; \
+		mv $@ $(CWD)
+
+.INTERMEDIATE: facebookImport.txt
+facebookImport.txt: ensure-rch
+	cd $(POLYPOD_FEATURES_PATH)/facebookImport; \
+		$(RCH) src/facebookImporter.jsx > $@; \
+		mv $@ $(CWD)
+
+.PHONY: ensure-rch
+ensure-rch: react-component-hierarchy
+	cd react-component-hierarchy; git pull --rebase; yarn install
+
+react-component-hierarchy:
```

Comment:
Precondition for install-deps too

Reply:
What do you mean by `install-deps`?

Reply:
Ah, you mean the _deps-install_ target you mentioned above? The `%.txt` files transitively depend on this one, because _ensure-rch_ depends on it.

Reply:
Right, but those should depend on install-deps or deps-install or whatever you want to call it. Essentially ensure that it meets all prerrequisites.

Reply:
That's not normally a thing in Makefiles, though there is a _setup_ target people sometimes have. But the philosophy is typically for people to install the required prerequisites themselves, e.g. Python in this case. I really think that's good enough for this case here, dealing with binary dependencies like this is pain.